From 6c4bc685cd48a704c19165c1bc0922ca65f3d6ff Mon Sep 17 00:00:00 2001
From: Quang Luong <qal@google.com>
Date: Wed, 9 Sep 2020 19:00:09 -0700
Subject: [PATCH 11/13] Don't prompt password for never-connected networks

Password shouldn't be prompted for secure saved networks that haven't
been connected to.

Bug: 160727383
Test: added a saved network via shell and manually connected to it, observe that
password prompt does not pop up.

Change-Id: Ib98fcbda077eacf22614aba766dba6ecb32a2ada
---
 .../src/com/android/wifitrackerlib/StandardWifiEntry.java   | 6 ------
 1 file changed, 6 deletions(-)

diff --git a/libs/WifiTrackerLib/src/com/android/wifitrackerlib/StandardWifiEntry.java b/libs/WifiTrackerLib/src/com/android/wifitrackerlib/StandardWifiEntry.java
index 19c2619bc..25eca2359 100644
--- a/libs/WifiTrackerLib/src/com/android/wifitrackerlib/StandardWifiEntry.java
+++ b/libs/WifiTrackerLib/src/com/android/wifitrackerlib/StandardWifiEntry.java
@@ -642,12 +642,6 @@ public class StandardWifiEntry extends WifiEntry {
             return false;
         }
 
-        // The secured Wi-Fi entry is never connected.
-        if (getSecurity() != SECURITY_NONE && getSecurity() != SECURITY_OWE
-                && !wifiConfig.getNetworkSelectionStatus().hasEverConnected()) {
-            return true;
-        }
-
         // The network is disabled because of one of the authentication problems.
         NetworkSelectionStatus networkSelectionStatus = wifiConfig.getNetworkSelectionStatus();
         if (networkSelectionStatus.getNetworkSelectionStatus() != NETWORK_SELECTION_ENABLED) {
-- 
2.35.1

